sdk-lib/src/main/rust/lib.rs

```diff
@@ -264,16 +258,15 @@ pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_tool_DerivationTool_deriveU
             return Err(format_err!("accountIndex argument must be positive"));
         };
 
-        let (di, sapling) = spending_key(&seed, network.coin_type(), AccountId(account_index))
-            .default_address()
-            .unwrap();
-        let p2pkh = utils::p2pkh_addr(
-            utils::p2pkh_full_viewing_key(&network, &seed, AccountId(account_index)).unwrap(),
-            di,
-        )
-        .unwrap();
-        // TODO: replace this with `zcash_address::unified::Address`.
-        let address_str = utils::fake_ua_encode(&p2pkh, &sapling);
+        let account_id = AccountId::from(account_index);
+        let ufvk = UnifiedSpendingKey::from_seed(&network, &seed, account_id)
+            .map_err(|e| format_err!("error generating unified spending key from seed: {:?}", e))
+            .map(|usk| usk.to_unified_full_viewing_key())?;
+
+        let (ua, _) = ufvk
+            .find_address(DiversifierIndex::new())
```

Comment:
I think that this should be `DiversifierIndex::zero()` or `DiversifierIndex::from(0)`; `new` doesn't seem like the right constructor for `DiversifierIndex`.

Reply:
I prefer the `orchard` `DiversifierIndex` type, but we're in the `zcash_*` crates so I used the one from `zcash_primitives`. We should probably figure out a way to reify the types.

Reply:
Maybe `DiversifierIndex` should be in `zcash_address`?